gooddata/gooddata-ui-sdk
# RELATED: RAIL-4192 Upgrade OpenAPITools

* upgrade to the latest CLI
* upgrade to the latest core
* tweak config to keep the breakage to a minimum
* remove not needed custom templates: this brings some improvements
  * utility function for checking required parameters (much shorter code)
  * better enums (not the expensive native ones, but lightweight object-based ones)
  * deprecated tag is now correctly propagated
* use the native useSingleRequestParameter parameter instead of our home made logic
  * this has a [known bug on upstream](https://github.com/OpenAPITools/openapi-generator/issues/6783), so we work around that tweaking our templates
  * it also enables us to use better TSDoc for the generated request parameters
  * it also properly types the options that allowed us to find several small bugs/hacks

The most outwards-breaking change is the unification of the enum value names: they are now all CONSTANT_CASE as it was the prevalent type before the changes (but was not used consistently). This should be easy to adapt to, though. Other that that the changes are improved TSDoc and types which should not break anyone.

JIRA: RAIL-4192

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)
